Add documentation to project that provides high level view...

Krishna opened the pull request:
… of code and corresponding UI components.

I have created the document using Sketch, and have supplied a PDF export of the file for easier consumption.

The document is by no means comprehensive, but I think it should help newcomers get up to speed with the codebase a bit more quickly.

boek commented:
Hey @Krishna, this is really awesome stuff!

My only concern with this would be the maintainability. While I love Sketch, I actually don't have a license to it. I personally have an affinity for Markdown (Which we can embed assets in as well), but I'm open to other suggestions as well.

Krishna commented:
Hi @boek. 

I understand about the restrictions of using Sketch - I just used it because it felt like the quickest tool to throw something visual together. 

Would a Pages or Keynote file be good replacement? The only reason I don't want to go down to Markdown or a HTML-solution is because this particular document is heavy on the visuals side (screenshots, with arrows calling out specific UI elements). This is a lot easier and quicker to create using a WYSIWYG tool with good image/drawing support.

I'm happy to consider some other tools if you have any suggestions. Thanks!

boek commented:
Hey @Krishna,

I definitely think we should keep the visuals! Those could be created in a users application of choice (maybe we could create a style guide for them?). I was mainly thinking for structure and the copy. 

Some other benefits to using Markdown on Github:
* Sections can be easily referenced in IRC/Tickets/PRs
* Diffs are easier to review between versions. 
* Code blocks for components

bkmunar commented:
Also, code will be changing quite a bit in the net few weeks with new features and refactoring. So it's best to maintain some sort of a "Valid up to this date" sort of note on the documentation. But this is great thank you!!

Krishna commented:
@boek @bkmunar I've updated the PR with a markdown formatted document, with the images split out into seperate files. I've also added a 'Revisions' section at the start of the document which indicates the date that a particular revision of the document was created, what changes were made and who authored the changes.

Let me know your thoughts!

bkmunar commented:
Looks great!! However, the links give me 404s.... @boek wanna double check? also sanity check the facts too? Thanks!

Krishna commented:
@boek - I made the changes you requested.

Krishna commented:
Hi,

Was wondering if there are any more changes required for this PR to get merged in?

boek commented:
Hey @Krishna, my apologies for the delay! This looks great :)
